### ISSUE TITLE
Support urls with underscores

### ISSUE THREAD

**Issue by Dock1100** [enqueue/metabase-clickhouse-driver#23]
On the Metabase db configuration page, if the host contains underscores, e.g.: `clickhouse_app`, connection attempt always results in `port is missed or wrong`. With hosts without underscores, it works like a correctly.

Metabase log:
```
08-09 07:59:08 DEBUG middleware.log :: PUT /api/database/4 400 10 ms (1 DB calls) 
{:valid false, :dbname "port is missed or wrong", :message "port is missed or wrong"}
```

Driver version: 0.3 https://github.com/enqueue/metabase-clickhouse-driver/releases/download/0.3/clickhouse.metabase-driver.jar
Metabase version: v0.32.10

**Comment by iusik** [comment]
How to solve this problem ?

**Comment by mcarbonneaux** [comment]
the problem seem to come from a bug/strictness in java.net.URI from jdk ...

java.net.URI are used in https://github.com/ClickHouse/clickhouse-jdbc/blob/master/clickhouse-jdbc/src/main/java/ru/yandex/clickhouse/ClickhouseJdbcUrlParser.java#L42 to decode jdbc url...

two bug are in jdk : https://bugs.openjdk.org/browse/JDK-8221675 and https://bugs.openjdk.org/browse/JDK-8019345.

also spring had the same pb... https://github.com/spring-projects/spring-framework/issues/24439

the good solution while be to fix in java.net.URI... 

the other solution are to use alternative URI parse in clickhouse jdbc driver...

**Comment by mcarbonneaux** [comment]
i'found the message "port is missed or wrong" are throw by old clickhouse-jdbc version....  an update of the clickhouse-jdbc dependancy can change a little the problem...

**Comment by slvrtrn** [comment]
Should be fixed in the upcoming 0.9.0 release as it will include JDBC driver `0.3.2-patch11` by default.

**Comment by slvrtrn** [comment]
Should be fixed in [0.9.0](https://github.com/enqueue/metabase-clickhouse-driver/releases/tag/0.9.0)
